hashing/src/chunk_with_proof.rs

```diff
@@ -51,24 +51,31 @@ impl ChunkWithProof {
     ///
     /// Empty data is always represented as single, empty chunk and not as zero chunks.
     pub fn new(data: &[u8], index: u64) -> Result<Self, MerkleConstructionError> {
-        let (proof, chunk) = if data.is_empty() {
-            (
+        let (proof, chunk) = match data.is_empty() {
+            true if index > 0 => {
+                return Err(MerkleConstructionError::IndexOutOfBounds { count: 1, index });
```

Comment:
Why is count one here?  Perhaps you want a new error variant?

Reply:
If I remember correctly we went with "emtpy data is one empty chunk, not zero chunks", so `count` is 1 and the only valid value for `index` is 0. I think it makes sense to use the same error variant here.

Reply:
This makes sense, thanks @afck !